File: frontend/src/types/minisite.ts
export interface SiteFeature {
  title: string;
  description?: string;
  icon?: string;
}

export interface SiteTestimonial {
  name: string;
  text: string;
  role?: string;
  avatar_url?: string;
}

export interface MiniSiteData {
  id: number;
  user_id: number;
  tenant_id: number | null;
  slug: string;
  status: 'draft' | 'published' | 'archived';
  template_id: string;
  site_name: string;
  tagline: string | null;
  logo_url: string | null;
  favicon_url: string | null;
  hero_title: string | null;
  hero_subtitle: string | null;
  hero_image_url: string | null;
  hero_cta_text: string | null;
  hero_cta_url: string | null;
  about_title: string | null;
  about_text: string | null;
  about_image_url: string | null;
  features: SiteFeature[];
  testimonials: SiteTestimonial[];
  gallery: string[];
  cta_title: string | null;
  cta_subtitle: string | null;
  cta_button_text: string | null;
  cta_button_url: string | null;
  phone: string | null;
  email: string | null;
  address: string | null;
  whatsapp: string | null;
  website: string | null;
  linkedin: string | null;
  instagram: string | null;
  github: string | null;
  twitter: string | null;
  youtube: string | null;
  footer_text: string | null;
  custom_colors: Record<string, string>;
  custom_fonts: Record<string, string>;
  view_count: number;
  created_at: string | null;
  updated_at: string | null;
}

export type MiniSiteCreatePayload = Omit<
  MiniSiteData,
  'id' | 'user_id' | 'tenant_id' | 'slug' | 'status' | 'view_count' | 'created_at' | 'updated_at'
>;

export type MiniSiteUpdatePayload = Partial<MiniSiteCreatePayload> & { status?: string };

export type MiniSitePublicData = Omit<
  MiniSiteData,
  'id' | 'user_id' | 'tenant_id' | 'status' | 'view_count' | 'created_at' | 'updated_at'
>;

File: backend/app/schemas/minisite.py
"""MiniSite Pydantic schemas."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class SiteFeature(BaseModel):
    title: str
    description: Optional[str] = None
    icon: Optional[str] = None


class SiteTestimonial(BaseModel):
    name: str
    text: str
    role: Optional[str] = None
    avatar_url: Optional[str] = None


class MiniSiteCreateRequest(BaseModel):
    template_id: str = "gradient-startup"
    site_name: str = Field(..., min_length=1, max_length=200)
    tagline: Optional[str] = None
    logo_url: Optional[str] = None
    favicon_url: Optional[str] = None
    hero_title: Optional[str] = None
    hero_subtitle: Optional[str] = None
    hero_image_url: Optional[str] = None
    hero_cta_text: Optional[str] = None
    hero_cta_url: Optional[str] = None
    about_title: Optional[str] = None
    about_text: Optional[str] = None
    about_image_url: Optional[str] = None
    features: list[SiteFeature] = []
    testimonials: list[SiteTestimonial] = []
    gallery: list[str] = []
    cta_title: Optional[str] = None
    cta_subtitle: Optional[str] = None
    cta_button_text: Optional[str] = None
    cta_button_url: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    address: Optional[str] = None
    whatsapp: Optional[str] = None
    website: Optional[str] = None
    linkedin: Optional[str] = None
    instagram: Optional[str] = None
    github: Optional[str] = None
    twitter: Optional[str] = None
    youtube: Optional[str] = None
    footer_text: Optional[str] = None
    custom_colors: dict = {}
    custom_fonts: dict = {}


class MiniSiteUpdateRequest(BaseModel):
    template_id: Optional[str] = None
    site_name: Optional[str] = None
    tagline: Optional[str] = None
    logo_url: Optional[str] = None
    favicon_url: Optional[str] = None
    hero_title: Optional[str] = None
    hero_subtitle: Optional[str] = None
    hero_image_url: Optional[str] = None
    hero_cta_text: Optional[str] = None
    hero_cta_url: Optional[str] = None
    about_title: Optional[str] = None
    about_text: Optional[str] = None
    about_image_url: Optional[str] = None
    features: Optional[list[SiteFeature]] = None
    testimonials: Optional[list[SiteTestimonial]] = None
    gallery: Optional[list[str]] = None
    cta_title: Optional[str] = None
    cta_subtitle: Optional[str] = None
    cta_button_text: Optional[str] = None
    cta_button_url: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    address: Optional[str] = None
    whatsapp: Optional[str] = None
    website: Optional[str] = None
    linkedin: Optional[str] = None
    instagram: Optional[str] = None
    github: Optional[str] = None
    twitter: Optional[str] = None
    youtube: Optional[str] = None
    footer_text: Optional[str] = None
    custom_colors: Optional[dict] = None
    custom_fonts: Optional[dict] = None
    status: Optional[str] = None


class MiniSiteResponse(BaseModel):
    id: int
    user_id: int
    tenant_id: Optional[int] = None
    slug: str
    status: str
    template_id: str
    site_name: str
    tagline: Optional[str] = None
    logo_url: Optional[str] = None
    favicon_url: Optional[str] = None
    hero_title: Optional[str] = None
    hero_subtitle: Optional[str] = None
    hero_image_url: Optional[str] = None
    hero_cta_text: Optional[str] = None
    hero_cta_url: Optional[str] = None
    about_title: Optional[str] = None
    about_text: Optional[str] = None
    about_image_url: Optional[str] = None
    features: list[SiteFeature] = []
    testimonials: list[SiteTestimonial] = []
    gallery: list[str] = []
    cta_title: Optional[str] = None
    cta_subtitle: Optional[str] = None
    cta_button_text: Optional[str] = None
    cta_button_url: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    address: Optional[str] = None
    whatsapp: Optional[str] = None
    website: Optional[str] = None
    linkedin: Optional[str] = None
    instagram: Optional[str] = None
    github: Optional[str] = None
    twitter: Optional[str] = None
    youtube: Optional[str] = None
    footer_text: Optional[str] = None
    custom_colors: dict = {}
    custom_fonts: dict = {}
    view_count: int = 0
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class MiniSitePublicResponse(BaseModel):
    slug: str
    template_id: str
    site_name: str
    tagline: Optional[str] = None
    logo_url: Optional[str] = None
    favicon_url: Optional[str] = None
    hero_title: Optional[str] = None
    hero_subtitle: Optional[str] = None
    hero_image_url: Optional[str] = None
    hero_cta_text: Optional[str] = None
    hero_cta_url: Optional[str] = None
    about_title: Optional[str] = None
    about_text: Optional[str] = None
    about_image_url: Optional[str] = None
    features: list[SiteFeature] = []
    testimonials: list[SiteTestimonial] = []
    gallery: list[str] = []
    cta_title: Optional[str] = None
    cta_subtitle: Optional[str] = None
    cta_button_text: Optional[str] = None
    cta_button_url: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    address: Optional[str] = None
    whatsapp: Optional[str] = None
    website: Optional[str] = None
    linkedin: Optional[str] = None
    instagram: Optional[str] = None
    github: Optional[str] = None
    twitter: Optional[str] = None
    youtube: Optional[str] = None
    footer_text: Optional[str] = None
    custom_colors: dict = {}
    custom_fonts: dict = {}

    class Config:
        from_attributes = True
